SignalsIntegrationTests/SignalsIntegrationTests/SignalDataTests.cs

```diff
@@ -22,7 +23,249 @@ public static new void ClassCleanup()
         }
 
         [TestMethod]
-        public void CanWriteAndRetrieveData()
+        public void CanWriteAndRetrieveSecondData()
+        {
+            var path = SignalPathGenerator.Generate();
+            var timestamp = new DateTime(2019, 4, 14);
+
+            var newSignal1 = new Signal()
+            {
+                Path = path,
+                Granularity = Granularity.Second,
+                DataType = DataType.Integer
+            };
+
+            var signal = client.Add(newSignal1.ToDto<Dto.Signal>());
+
+            var data = new[]
+            {
+                new Datum<int>()
+                {
+                    Timestamp = timestamp,
+                    Value = 4,
+                    Quality = Quality.Fair
+                }
+            };
+
+            client.SetData(signal.Id.Value, data.ToDto<Dto.Datum[]>());
+            var retrievedData = client.GetData(signal.Id.Value, timestamp, timestamp.AddSeconds(1))
+                .ToDomain<Domain.Datum<int>[]>();
+
+            CollectionAssert.AreEqual(data, retrievedData, new DatumComparer<int>());
+        }
+
+        [TestMethod]
+        public void CanWriteAndRetrieveSecondDataIgnoringOrder()
+        {
+            var path = SignalPathGenerator.Generate();
+            var timestamp = new DateTime(2019, 4, 14);
+
+            var newSignal1 = new Signal()
+            {
+                Path = path,
+                Granularity = Granularity.Second,
+                DataType = DataType.Integer
+            };
+
+            var signal = client.Add(newSignal1.ToDto<Dto.Signal>());
+
+            var data = new[]
+            {
+                new Datum<int>()
+                {
+                    Timestamp = timestamp.AddSeconds(1),
+                    Value = 5,
+                    Quality = Quality.Fair
+                },
+                new Datum<int>()
+                {
+                    Timestamp = timestamp,
+                    Value = 4,
+                    Quality = Quality.Fair
+                }
+            };
+
+            client.SetData(signal.Id.Value, data.ToDto<Dto.Datum[]>());
+            var retrievedData = client.GetData(signal.Id.Value, timestamp, timestamp.AddSeconds(2))
+                .ToDomain<Domain.Datum<int>[]>();
+
+            CollectionAssert.AreEqual(data.OrderBy(d => d.Timestamp).ToArray(), retrievedData, new DatumComparer<int>());
+        }
+
+        [TestMethod]
+        public void CanWriteAndRetrieveMinuteData()
+        {
+            var path = SignalPathGenerator.Generate();
+            var timestamp = new DateTime(2019, 4, 14);
+
+            var newSignal1 = new Signal()
+            {
+                Path = path,
+                Granularity = Granularity.Minute,
```

Comment:
so, ctrl+cv instead of parametrized tests? :D 
to close #25 I need to look through all hardcoded granularities (but, yes, it seems it fixes it :) )